server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/api/posttask/PostProjectAnalysisTasksExecutor.java

```diff
@@ -96,27 +99,32 @@ public void finished(boolean allStepsExecuted) {
       return;
     }
 
-    ProjectAnalysis projectAnalysis = createProjectAnalysis(allStepsExecuted ? SUCCESS : FAILED);
+    ProjectAnalysisImpl projectAnalysis = createProjectAnalysis(allStepsExecuted ? SUCCESS : FAILED);
     for (PostProjectAnalysisTask postProjectAnalysisTask : postProjectAnalysisTasks) {
       executeTask(projectAnalysis, postProjectAnalysisTask);
     }
   }
 
-  private static void executeTask(ProjectAnalysis projectAnalysis, PostProjectAnalysisTask postProjectAnalysisTask) {
+  private static void executeTask(ProjectAnalysisImpl projectAnalysis, PostProjectAnalysisTask postProjectAnalysisTask) {
     try {
       postProjectAnalysisTask.finished(projectAnalysis);
     } catch (Exception e) {
       LOG.error("Execution of task " + postProjectAnalysisTask.getClass() + " failed", e);
     }
   }
 
-  private ProjectAnalysis createProjectAnalysis(CeTask.Status status) {
+  private ProjectAnalysisImpl createProjectAnalysis(CeTask.Status status) {
     Long analysisDate = getAnalysisDate();
 
-    return new ProjectAnalysis(
+    AnalysisImpl analysis = null;
```

Comment:
please fix this by replacing `getAnalysisDate()` method (call at above line) by a method `getAnalysis()'